nautobot_circuit_maintenance/models.py

```diff
@@ -190,6 +194,15 @@ def to_csv(self):
         """Return fields for bulk view."""
         return (self.name, self.slug, self.providers)
 
+    def set_token(self, value):
+        """Getter for _token."""
+        self._token = pickle.dumps(value)
+        self.save()
+
+    def get_token(self):
+        """Setter for _token."""
+        return pickle.loads(self._token)  # nosec
+
```

Comment:
Might be more Pythonic to use a `property` here, i.e.:

```python
@property
def token(self):
     return pickle.loads(self._token)

@token.setter
def token(self, value):
    self._token = pickle.dumps(value)
```

Also I wouldn't have the setter function automatically call `self.save()` as that doesn't match the general model API - would be better to explicitly call `save()` as needed.

Reply:
totally, it's more pythonic and also moved `save()` to the caller.